idaes_examples/notebooks/docs/unit_models/custom_unit_models/Aq_property.py

```diff
@@ -0,0 +1,306 @@
+# Changes the divide behavior to not do integer division
+from __future__ import division
+
+# Import Python libraries
+import logging
+
+import idaes.logger as idaeslog
+from idaes.core.util.initialization import fix_state_vars, revert_state_vars
+
+# Import Pyomo libraries
+from pyomo.environ import (
+    Param,
+    Var,
+    NonNegativeReals,
+    units,
+    Expression,
+    PositiveReals,
+)
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialFlowBasis,
+    PhysicalParameterBlock,
+    StateBlockData,
+    StateBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    Solute,
+    Solvent,
+    LiquidPhase,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+# Some more information about this module
+__author__ = "Javal Vyas"
+
+
+# Set up logger
+_log = logging.getLogger(__name__)
+
+
+@declare_process_block_class("AqPhase")
+class PhysicalParameterData(PhysicalParameterBlock):
+    """
+    Property Parameter Block Class
+
+    Contains parameters and indexing sets associated with properties for
+    liquid Phase
+
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super(PhysicalParameterData, self).build()
+
+        self._state_block_class = AqPhaseStateBlock
+
+        # List of valid phases in property package
+        self.Aq = LiquidPhase()
+
+        # Component list - a list of component identifiers
+        self.NaCl = Solute()
+        self.KNO3 = Solute()
+        self.CaSO4 = Solute()
+        self.H2O = Solvent()
+
+        # Heat capacity of solvent
+        self.cp_mass = Param(
+            mutable=True,
+            initialize=4182,
+            doc="Specific heat capacity of solvent",
+            units=units.J / units.kg / units.K,
+        )
+
+        self.dens_mass = Param(
+            mutable=True,
+            initialize=997,
+            doc="Density of ethylene dibromide",
+            units=units.kg / units.m**3,
+        )
+        self.temperature_ref = Param(
+            within=PositiveReals,
+            mutable=True,
+            default=298.15,
+            doc="Reference temperature",
+            units=units.K,
+        )
+
+    @classmethod
+    def define_metadata(cls, obj):
+        obj.add_properties(
+            {
+                "flow_mol": {"method": None, "units": "kmol/s"},
```

Comment:
These units are not consistent with the base units defined below. IDAES works on the basis that all outward facing properties must have units based on those listed in the default units metadata so that we know how to convert when unit conversions are required. Whilst this sometimes means we end up with different units than we might prefer, this is necessary to make things work efficiently.

On a second note, the `"units"` entry here is no longer used and should be removed.